src/H5FDsec2.c

```diff
@@ -700,15 +700,16 @@ H5FD__sec2_read(H5FD_t *_file, H5FD_mem_t H5_ATTR_UNUSED type, hid_t H5_ATTR_UNU
             int    myerrno = errno;
             time_t mytime  = time(NULL);
 
+#ifndef H5_HAVE_PREADWRITE
             offset = HDlseek(file->fd, 0, SEEK_CUR);
```

Comment:
Since `pread` doesn't change the file offset, this `HDlseek` call ends up resetting the file offset that gets printed out on errors.